src/model/activity.h

```diff
@@ -10,17 +10,20 @@
 namespace gologpp {
 
 
-class Activity : public Grounding<Action>, public LanguageElement<Activity> {
+class Activity
+: public Grounding<Action>
+, public LanguageElement<Activity>
+, public std::enable_shared_from_this<Activity> {
 public:
 	enum State { IDLE, RUNNING, FINAL, PREEMPTED, FAILED };
 
-	Activity(const shared_ptr<Action> &action, vector<unique_ptr<Value>> &&args, State state = IDLE);
-	Activity(const shared_ptr<Transition> &);
+	Activity(const shared_ptr<Action> &action, vector<unique_ptr<Value>> &&args, AExecutionContext &, State state = IDLE);
+	Activity(const shared_ptr<Transition> &, AExecutionContext &);
 
 	State state() const;
 	void set_state(State s);
 
-	shared_ptr<Transition> transition(Transition::Hook hook);
+	void update(Transition::Hook hook, Value *sensing_result = nullptr);
```

Comment:
I'd use `std::optional` instead of using a `nullptr` to indicate that there is no sensing result.

Reply:
Yeah, I'd certainly like to, but that's C++17 and I have to be careful with that. Ubuntu 16.04, remember? But I'll check, maybe their libstdc++ can do it ;-)

Reply:
On the other hand, we already have a dependency on `boost::optional` because of the parser, so we might just go with that.

Reply:
Oh I thought it was C++14, but you're right. `boost::optional` works fine, too!

Reply:
done.